packages/daemon/src/locator.js

```diff
@@ -28,48 +28,50 @@ const isValidLocatorType = allegedType =>
  */
 const assertValidLocatorType = allegedType => {
   if (!isValidLocatorType(allegedType)) {
-    assert.Fail`Unrecognized locator type ${q(allegedType)}`;
+    throw assert.error(`Unrecognized locator type ${q(allegedType)}`);
```

Comment:
Linter trouble you about type narrowing? These should be equivalent.

Reply:
They are equivalent, I just found older uses of `throw assert.error()` in `formula-identifier.js` and decided to standardize on that more familiar (to me) idiom.

As of this change, there are only two other uses of `assert.Fail` in the daemon. We could make all of these cases use the same thing?

Reply:
Agoric style favors `assert.Fail` and I just haven’t gone back to make the dæmon consistent with the emerging æsthetic.

Reply:
Ah, there was one other reason: TypeScript does not appear to understand that `assert.Fail` throws, and so some of my type guards only actually narrow the types when using `throw assert.error()` as opposed to `assert.Fail`.

Reply:
I’m seeing the same. We can circle back to fix it.